### PR TITLE
Push notification event broadcast

### PR DESCRIPTION
**Pull request by paulwsmith**
## Push notification event broadcasts
Adds event broadcasts when Snapyr push notifications are received or tapped. A consumer can set up a broadcast receiver to listen to these and run custom code when a push notification event occurs. This enables callbacks in the React Native SDK, but could also be listened to directly by other Android code.

A lot of the code changes in this PR involve moving logic from `SnapyrNotificationListener` and `SnapyrFirebaseMessagingService` into a new class, `SnapyrNotification`. This is a structured representation of a notification object. It implements the Parcelable interface so that it can be passed around as part of an Intent, and reconstituted back to a full `SnapyrNotification` object on the receiving side.

## Debugging improvements
Updates the Gradle config to allow this project to be included as a local subproject, within another Gradle project. This facilitates development of React Native and other SDKs that use this SDK, without needing to perform a full build and `.aar` update after every code change.

This is optional, and doesn't change how to work with this project on its own.

## Bug fix
Incidental fix for a bug where only one notification for an app would remain in the notification drawer, if multiple notifications were received while the application is in a closed state, due to the notification ID counter getting reset to 0 each time.

### CODE REVIEW

File: snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrNotification.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+import java.util.Map;
+import java.util.Random;
+import org.json.JSONObject;
+
+public class SnapyrNotification implements Parcelable {
+    private static final Random random =
+            new Random(); // seeded with current System.nanotime() by default
+
+    public final int notificationId =
+            random.nextInt(Integer.MAX_VALUE); // MAX_VALUE ensures non-negative
```
Comment (onemorebsmith) [inline snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrNotification.java:46]: Weird.

File: snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrNotification.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+        String deepLinkUrl =
+                getOrDefault(jsonData, SnapyrNotificationHandler.ACTION_DEEP_LINK_KEY, null);
+        this.deepLinkUrl = (deepLinkUrl != null) ? Uri.parse(deepLinkUrl) : null;
+
+        this.imageUrl = getOrDefault(jsonData, SnapyrNotificationHandler.NOTIF_IMAGE_URL_KEY, null);
+    }
+
+    protected SnapyrNotification(Parcel in) {
+        titleText = in.readString();
```
Comment (onemorebsmith) [inline snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrNotification.java:141]: maybe a comment here stating that the order matters and must match the parcel out?

**Review by onemorebsmith — DISMISSED** [review]
Great stuff @paulwsmith . Going to start calling you the gradle wizard

File: snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrNotification.java (hunk 2 above)
Comment (paulwsmith) [reply snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrNotification.java:141]: Good call, will add that

**Pull request merged**